oauth2/oauth2-jose/src/test/java/org/springframework/security/oauth2/jwt/NimbusJwtDecoderTests.java

```diff
@@ -346,6 +347,23 @@ public void decodeWhenSignatureMismatchesAlgorithmThenThrowsException() throws E
 				.isInstanceOf(BadJwtException.class);
 	}
 
+	// gh-8730
+	@Test
+	public void decodeWhenCustomizingJwtProcessor() throws Exception {
+		RSAPublicKey publicKey = TestKeys.DEFAULT_PUBLIC_KEY;
+		RSAPrivateKey privateKey = TestKeys.DEFAULT_PRIVATE_KEY;
+		JWSHeader header = new JWSHeader.Builder(JWSAlgorithm.RS256).type(new JOSEObjectType("custom")).build();
```

Comment:
Is your use case that `typ` is set to `JWS`? If so, you might consider changing the test to that to prove that it'll work for your use case.